<?xml version="2.0"?>
<launch>
   <!-- launch video stream -->
   <include file="$(find video_stream_opencv)/launch/camera.launch" >
                <!-- node name and ros graph name -->
                <arg name="camera_name" value="ecam" />
                <!-- means video device 0, /dev/video0 -->
                <arg name="video_stream_provider" value="1" />
                <!-- set camera fps to (if the device allows) -->
                <arg name="set_camera_fps" value="30"/>
        <!-- set buffer queue size of frame capturing to (1 means we want the latest frame only) -->
        <arg name="buffer_queue_size" value="1" />
                <!-- throttling the querying of frames to -->
                <arg name="fps" value="30" />
                <!-- setting frame_id -->
                <arg name="frame_id" value="ecam_left" />
                <!-- camera info loading, take care as it needs the "file:///" at the start , e.g.:
                "file:///$(find your_camera_package)/config/your_camera.yaml" -->
                <arg name="camera_info_url" value="" />
                <!-- flip the image horizontally (mirror it) -->
                <arg name="flip_horizontal" value="false" />
                <!-- flip the image vertically -->
                <arg name="flip_vertical" value="false" />
                <!-- force width and height, 0 means no forcing -->
        
            <arg name="width" value="1280"/>
            <arg name="height" value="720"/>

                <!-- visualize on an image_view window the stream generated -->
                <arg name="visualize" value="false" />
   </include>

    <rosparam file="$(find boat_vision)/config/logitech_config.yaml" command="load"/>

    <node pkg="boat_vision" type="ecam_info_node" name="ecam_info_node"/>
   <node pkg="boat_vision" type="ecam_undistort_node" name="ecam_undistort_node"/>
</launch>